shopAPI > order > GuestOrder, OrderConfiguration API  작성

LeeYhunHwan opened the pull request:
## guestOrder
- 비회원 장바구니 계산하여 갖고오기 (getShoppingCart)
- 비회원 주문 상세 조회하기 (orderDetail)
- 비회원 주문 토큰 발급하기 (issueOrderToken)
- 비회원 상품 주문 구매확정 처리하기 (purchaseConfirm)
- 비회원 상품 주문 배송완료 처리하기 (deliveryCompleted)
- 비회원 현금영수증 신청하기 (receiptRequest)
- 비회원 주문 상세 조회하기(클레임 상세사유 포함) (orderCancelDetail)
- 비회원 주문단위 배송정보 수정하기 (modifyDeliveryInfo)
- 비회원 초기화된 주문 패스워드 전송하기 (getPasswordByEmail)
## orderConfiguration
- 주문 설정 값 가져오기 (orderConfigs)

LeeYhunHwan committed:
:sparkles: feat: shopAPI > order > guestOrder,orderConfiguration API  작성

guestOrder > 비회원 장바구니 계산하기
guestOrder > 비회원 주문 상세 조회하기
guestOrder > 비회원 주문 토큰 발급하기
guestOrder > 비회원 상품 주문 구매확정 처리하기
guestOrder > 비회원 상품 주문 배송완료 처리하기
guestOrder > 비회원 현금영수증 신청하기
guestOrder > 비회원 주문 상세 조회하기(클레임 상세사유 포함)
guestOrder > 비회원 주문단위 배송정보 수정하기
guestOrder > 비회원 초기화된 주문 패스워드 전송하기
orderConfiguration > 주문 설정 값 가져오기

jerome-geek commented:
@LeeYhunHwan 함수명 전체적으로 다시 한 번 확인해주세요.
e.g.) 비회원 장바구니 계산하여 가져오기 -> getShoppingCart면 회원 장바구니를 가져올 수 있다고도 생각할 수 있을 것 같고, Cart를 굳이 ShoppingCart로 쓸 필요가 있을까요?
비회원 주문 상세 조회하기 -> 함수명은 동사로 시작하는게 좋아요, 조회하기 기능인데 orderDetail은 전혀 맞지 않겠죠?

전체적으로 다시 한 번 확인 부탁드립니다

LeeYhunHwan committed:
:ok_hand: fix: guestOrder.ts 함수명 변경

- getShoppingCart -> getCart
- orderDetail -> getOrderDetail
- deliveryCompleted -> requestDeliveryCompletion
- receiptRequest -> requestReceipt
- orderCancelDetail -> getOrderCancelDetail
- modifyDeliveryInfo -> updateDeliveryInfo

LeeYhunHwan committed:
:hammer: fix: guestOrder API 타입 값 수정 및 parameter intersection

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,169 @@
+            data: { guestToken, cashReceiptIssuePurposeType, cashReceiptKey },
+        }),
+
+    // TODO GuestToken, orderNo 모름, 400 error 발생 추후 테스트 필요
+    getOrderCancelDetail: (
+        orderNo: String,
+        {
+            orderRequestType,
+            claimType,
+        }: { orderRequestType?: ORDER_REQUEST_TYPE; claimType?: CLAIM_TYPE },
+    ): Promise<AxiosResponse> =>
+        request({
+            method: 'GET',
+            url: `/guest/orders/${orderNo}/claim`,
+            params: { orderRequestType, claimType },
+        }),
```
이 함수는 취소 주문만 가져오는건가요?

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,169 @@
+        }),
+
+    // TODO orderNo 모름, 400 error 발생 추후 테스트 필요
+    getPasswordByEmail: (
+        orderNo: String,
+        { replyType, mobileNo, email, name }: PasswordParams,
+    ): Promise<AxiosResponse> =>
+        request({
+            method: 'GET',
+            url: `guest/orders/${orderNo}/forgot-password`,
+            params: { replyType, mobileNo, email, name },
+        }),
+};
```
return 값이 없는데 get이라는 동사를 사용하기엔 적절하지 않은 것 같은데 어떻게 생각하시나요?

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,169 @@
+import { AxiosResponse } from 'axios';
+
+import request from 'api/core';
+import { defaultHeaders } from 'api/core';
```
요건 한 줄로 쓰는게 좋겠죠?

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,169 @@
+            data: { password, name, mobileNo, email, orderRequestType },
+        }),
+
+    // TODO GuestToken, orderOptionNo 모름, 400 error 발생 추후 테스트 필요
+    purchaseConfirm: (
+        orderOptionNo: String,
+        { guestToken }: GuestToken,
+    ): Promise<AxiosResponse> =>
+        request({
+            method: 'PUT',
+            url: `/guest/order-options/${orderOptionNo}/confirm`,
+            data: { guestToken },
+        }),
+
+    // TODO GuestToken, orderOptionNo 모름, 400 error 발생 추후 테스트 필요
+    requestDeliveryCompletion: (
+        orderOptionNo: String,
+        { guestToken }: GuestToken,
+    ): Promise<AxiosResponse> =>
+        request({
+            method: 'PUT',
+            url: `/guest/order-options/${orderOptionNo}/delivery-done`,
+            data: { orderOptionNo },
+            headers: Object.assign({}, defaultHeaders(), {
+                guestToken: guestToken,
+            }),
+        }),
```
함수의 역할이 `처리하기`인데 request는 맞지 않는 것 같아요, 그리고 headers의 guestToken은 shorthand로 써주시는 게 좋을 것 같아요

LeeYhunHwan committed:
:hammer: fix: guestOrder API 함수명 변경

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,168 @@
+        }),
+
+    // TODO GuestToken, orderNo 모름, 400 error 발생 추후 테스트 필요
+    getNonMemberOrderDetail: (
```
NonMember -> Guest가 낫지 않을까요?

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,168 @@
+        }),
+
+    // TODO GuestToken, orderOptionNo 모름, 400 error 발생 추후 테스트 필요
+    purchaseConfirm: (
```
구매확정이라면 confirmOrder정도로 쓸 수 있을 것 같습니다. purchase라는 동사는 함수 설명이랑 맞지 않죠?

LeeYhunHwan committed:
:ok_hand: fix: guestOrder API 함수명 수정

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,168 @@
+        request({
+            method: 'POST',
+            url: `/guest/orders/${orderNo}/cashReceipt`,
+            data: { guestToken, cashReceiptIssuePurposeType, cashReceiptKey },
```
guestToken은 body가 아니라 header에 들어가는 값이네요

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,168 @@
+        request({
+            method: 'PUT',
+            url: `/guest/order-options/${orderOptionNo}/confirm`,
+            data: { guestToken },
```
guestToken은 파라미터로 받지않고, body가 아니라 header에 들어가는 값이네요

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,168 @@
+            url: '/guest/cart',
+            data: [
+                {
+                    divideInvalidProducts,
```
divideInvalidProducts 는 queryString으로 들어가는 값이네요

jerome-geek replied:
@LeeYhunHwan 여기 수정이 안된 것 같아요

jerome-geek commented on src/api/order/guestOrder.ts:
```diff
@@ -0,0 +1,168 @@
+            ],
+        }),
+
+    getOrderDetail: (
+        orderNo: String,
+        { orderRequestType }: { orderRequestType?: ORDER_REQUEST_TYPE },
+    ): Promise<AxiosResponse> =>
+        request({
+            method: 'GET',
+            url: `/guest/orders/${orderNo}`,
+            params: { orderRequestType },
+        }),
```
여기는 왜 guestToken이 빠졌을까요?

jerome-geek replied:
@LeeYhunHwan  여기도 수정이 안된 것 같아요

jerome-geek reviewed (changes requested):
일단 guestToken은 header에 넣어주시고, accessToken과 똑같이 localStorage에서 가져오는 방식으로 작성 부탁드려요

LeeYhunHwan committed:
:ok_hand: fix: guestOrder API > guestToken 수정

LeeYhunHwan committed:
:ok_hand: fix: guestOrder API 수정
